src/main/scala/org/apache/spark/sql/delta/storage/BaseExternalLogStore.scala

```diff
@@ -90,23 +103,31 @@ abstract class BaseExternalLogStore(sparkConf: SparkConf, hadoopConf: Configurat
    * Method try to rewrite TransactionLog from temporary path if it not exists.
    * Method returns completed [[LogEntryMetadata]]
    */
-
-   private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
+  private def tryFixTransactionLog(fs: FileSystem, entry: LogEntryMetadata): LogEntryMetadata = {
     logDebug(s"Try to fix: ${entry.path}")
-    val tempPath = entry.tempPath.get
     try {
-      writeActions(fs, entry.path, read(tempPath).iterator)
-      val completedEntry = entry.complete()
-      writeCache(fs, completedEntry, overwrite = true)
-      deleteFile(fs, tempPath)
-      completedEntry
+      writeLogTransaction(fs, entry)
+    } catch {
+      case e: FileSystemException => print(
+        s"Failed to fix transaction, ignore... ${e.getMessage}\n"
+      )
+    }
+    val completedEntry = entry.complete()
+    writeCache(fs, completedEntry, overwrite = true)
+    completedEntry
+  }
+
+  private def writeLogTransaction(fs: FileSystem, entryMetadata: LogEntryMetadata) {
+    try {
+      copyFile(fs, entryMetadata.tempPath.get, entryMetadata.path)
+      writeCache(fs, entryMetadata.complete(), overwrite = true)
     } catch {
       case e: FileNotFoundException =>
```

Comment:
as we never removing temp files now maybe we should remove this try/catch block?

Reply:
If we accept failure of this operation, we should use try-catch block

Reply:
but this FileNotFoundException shouldn't be raised as we don't remove temp files, right?